znncli_general.go

```diff
@@ -0,0 +1,234 @@
+		}
+
+		// TODO handle decimal amounts
+		amount := big.NewInt(0)
+		amount, ok := amount.SetString(cCtx.Args().Get(1), 10)
+		if !ok {
+			fmt.Println("Error bad amount")
+			return nil
+		}
+
+		amount = amount.Mul(amount, big.NewInt(10^int64(token.Decimals)))
+
+		tmpl := template.Send(z.ProtocolVersion(), z.ChainIdentifier(), toAddress, zts, amount, []byte{})
+		_, err = utils.Send(z, tmpl, kp, false)
+		if err != nil {
+			fmt.Println("Error sending tx", err)
+			return err
+		}
+
+		return nil
+	},
+}
+
+var znnCliUnreceived = &cli.Command{
+	Name:  "unreceived",
+	Usage: "",
+	Action: func(cCtx *cli.Context) error {
+		if cCtx.NArg() != 0 {
+			fmt.Println("Incorrect number of arguments. Expected:")
+			fmt.Println("unreceived")
+			return nil
+		}
+
+		kp, err := getZnnCliSigner(walletDir, cCtx)
+		if err != nil {
+			fmt.Println("Error getting signer:", err)
+			return err
+		}
+		z, err := connect(url, chainId)
+		if err != nil {
+			fmt.Println("Error connecting to Zenon Network:", err)
+			return err
+		}
+
+		unreceived, err := z.Ledger.GetUnreceivedBlocksByAddress(kp.Address(), 0, 5)
+		if err != nil {
+			fmt.Println("Error fetching unreceived txs:", err)
+			return err
+		}
+		if len(unreceived.List) == 0 {
+			fmt.Println("Nothing to receive")
+			return nil
+		} else {
+			if unreceived.More {
+				fmt.Println("You have more than", unreceived.Count, "transaction(s) to receive")
+			} else {
+				fmt.Println("You have", unreceived.Count, "transaction(s) to receive")
+			}
+		}
+		fmt.Println("Showing the first", unreceived.Count)
+		for _, block := range unreceived.List {
+			fmt.Println("Unreceived", formatAmount(block.Amount, block.TokenInfo.Decimals), block.TokenInfo.TokenSymbol, "from", block.Address, "Use the hash", block.Hash, "to receive")
+		}
+		return nil
+	},
+}
+
+var znnCliReceiveAll = &cli.Command{
+	Name:  "receiveAll",
+	Usage: "",
+	Action: func(cCtx *cli.Context) error {
+		if cCtx.NArg() != 0 {
+			fmt.Println("Incorrect number of arguments. Expected:")
+			fmt.Println("receiveAll")
+			return nil
+		}
+
+		kp, err := getZnnCliSigner(walletDir, cCtx)
+		if err != nil {
+			fmt.Println("Error getting signer:", err)
+			return err
+		}
+		z, err := connect(url, chainId)
+		if err != nil {
+			fmt.Println("Error connecting to Zenon Network:", err)
+			return err
+		}
+
+		unreceived, err := z.Ledger.GetUnreceivedBlocksByAddress(kp.Address(), 0, 5)
+		if err != nil {
+			fmt.Println("Error fetching unreceived txs:", err)
+			return err
+		}
+		if len(unreceived.List) == 0 {
+			fmt.Println("Nothing to receive")
+			return nil
+		} else {
+			if unreceived.More {
+				fmt.Println("You have more than", unreceived.Count, "transaction(s) to receive")
+			} else {
+				fmt.Println("You have", unreceived.Count, "transaction(s) to receive")
+			}
+		}
+		fmt.Println("Please wait ...")
+
+		for unreceived.Count > 0 {
+			for _, block := range unreceived.List {
+				temp := template.Receive(1, uint64(chainId), block.Hash)
```

Comment:
```suggestion
				temp := template.Receive(z.ProtocolVersion(), z.ChainIdentifier(), block.Hash)
```
Otherwise will throw this error when running with the -hq flag:
```
nomctl -hq znn-cli -p "Don'tTrust.Verify" receiveAll
Using the default keyStore z1qqzsrnzpfev0dvu5eqge2t8kx4uzpmq30hpzpr
You have 1 transaction(s) to receive
Please wait ...
Error receiving txs: the block has a different network Id (1) from the node (321)
```

Reply:
thank you! i updated go-zdk for this https://github.com/hypercore-one/go-zdk/commit/b71304db9eb254b04fd38293e5652c8920a9c702
but forgot to update the usage here